docs/src/user-guide/hot_reload.md

```diff
@@ -0,0 +1,92 @@
+# Hot Reload
+
+Hot reload enables zero-downtime updates by transferring network listeners between Shotover instances. When you start a new Shotover instance with hot reload enabled, it requests the listening TCP sockets from the running instance. Once they are received, the new instance immediately starts accepting connections. Once the new shotover instance is fully operational, it sends a shutdown request to the old shotover instace and the original instance gradually drains its existing connections before shutting down. This allows you to deploy configuration changes or upgrade Shotover versions without dropping client connections.
+
+
+## How It Works
+
+When a new Shotover instance starts with hot reload enabled, it checks if a Unix socket exists at the configured path. If the socket exists, the new Shotover starts in hot reload mode and attempts to hot reload from the old Shotover instance.
+
+The hot reload process:
+
+1. The new instance connects to the Unix socket and requests the listening TCP socket file descriptors.
+
+2. The original instance transfers the file descriptors over the Unix socket using SCM_RIGHTS which is a Unix mechanism for sharing file descriptors between processes. After transferring, the original instance stops accepting new connections but continues to serve its existing connections.
+
+3. The new instance recreates the TCP listeners from the file descriptors received from the old shotover instance. These listeners uses the same IP addresses and ports as before.
+
+4. The new instance begins accepting new connections. At this point, the old instance still handles the existing connections. Once the new instance is fully functional, it sends a gradual shutdown request over the Unix socket to the old instance, specifying the shutdown duration.
+
+5. The original instance drains its existing connections gradually in chunks (see Gradual Shutdown below).
+
+6. Once all connections are drained, the original instance terminates and removes the Unix socket.
+
+7. Once the old instance terminates, the new instance will be able to create its own Unix socket. Once this is done, the new instance will be ready to be hot reloaded in the future.
```

Comment:
Instead of saying what shotover is "able" to do, we should say what it does.
Also, we should specify how the socket path is determined.

e.g.
`the new instance will be able to create its own Unix socket` -> `the new instance will create its own Unix Socket at the configured path.`